### PR TITLE
feat(fa): Update you own settings

### PR DESCRIPTION
**Pull request by MargretFinnboga**
# Update you own settings

https://app.asana.com/0/1201427499183230/1201234405372296

## What

Allows user to update their own information 
Adding pseudonymName to db 
Feature to turn on and off pseudonym name for applications 

## Why

Requested from staff members

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/financial-aid/web-veita/src/routes/MySettings/MySettings.tsx [hunk 1]
```diff
@@ -1,39 +1,62 @@
...
+    fetchPolicy: 'no-cache',
+    errorPolicy: 'all',
+  })
 
-  const [pseudonyms, setPseudonyms] = useState<boolean>(false)
+  useEffect(() => {
+    if (data && setAdmin) {
+      setAdmin(data.currentUser)
+    }
+  }, [data])
```
Comment (MargretFinnboga) [inline apps/financial-aid/web-veita/src/routes/MySettings/MySettings.tsx:52]: Im open for suggestions if there is a better solution for this 👐

File: apps/financial-aid/web-veita/src/routes/MySettings/MySettings.tsx [hunk 2]
```diff
@@ -1,39 +1,62 @@
...
+  const { admin, setAdmin } = useContext(AdminContext)
+
+  const [updateStaff, { loading }] = useMutation(UpdateStaffMutation)
+
+  const [getCurrentUser, { data }] = useLazyQuery<{
+    currentUser: User
+  }>(CurrentUserQuery, {
+    fetchPolicy: 'no-cache',
+    errorPolicy: 'all',
+  })
```
Comment (Rikkiabb) [inline apps/financial-aid/web-veita/src/routes/MySettings/MySettings.tsx:46]: On this screen we should not have the option to update nationalId and name, so let's start taking that out, after that is done then you can return staff from updateStaff

File: apps/financial-aid/web-veita/src/routes/MySettings/MySettings.tsx [hunk 3]
```diff
@@ -96,13 +73,45 @@ export const MySettings = () => {
...
+    !admin || !admin.staff || !state.email || !isEmailValid(state.email)
+
+  const onSubmitUpdate = async () => {
+    if (areRequiredFieldsFilled) {
+      setState({ ...state, hasError: true })
+      return
+    }
+
+    try {
+      await updateStaff({
```
Comment (Rikkiabb) [inline apps/financial-aid/web-veita/src/routes/MySettings/MySettings.tsx:93]: should we use the useStaff hook here?

File: apps/financial-aid/web-veita/src/routes/MySettings/MySettings.tsx [hunk 4]
```diff
@@ -155,9 +164,9 @@ export const MySettings = () => {
                   Nota dulnefni fyrir nöfn umsækjenda í yfirlitsskjám
                 </Text>
               }
-              checked={pseudonyms}
+              checked={state.usePseudoName}
               onChange={(newChecked) => {
-                setPseudonyms(newChecked)
+                setState({ ...state, usePseudoName: newChecked })
```
Comment (Rikkiabb) [inline apps/financial-aid/web-veita/src/routes/MySettings/MySettings.tsx:169]: Nitpick, but do we want to rename the `newChecked`variable here?

**Review by Rikkiabb — COMMENTED** [review]
Nits 🤏

**Review by Rikkiabb — APPROVED** [review]
🔥 💣 🚒 🧯

**Pull request merged**